src/nbl/video/IOpenGL_PhysicalDeviceBase.h

```diff
@@ -237,15 +237,81 @@ class IOpenGL_PhysicalDeviceBase : public IOpenGLPhysicalDeviceBase
+		assert(hasInString("ATI Technologies Inc.", "ati technologies", true));
+
 		// initialize features
-		std::string vendor = reinterpret_cast<const char*>(GetString(GL_VENDOR));
-		m_glfeatures.isIntelGPU = (vendor.find("Intel") != vendor.npos || vendor.find("INTEL") != vendor.npos);
+		const char* vendor = reinterpret_cast<const char*>(GetString(GL_VENDOR));
+		const char* renderer = reinterpret_cast<const char*>(GetString(GL_RENDERER));
+		const char* ogl_ver_str = reinterpret_cast<const char*>(GetString(GL_VERSION));
+
+		// Detecting DriverID from vendor, renderer and gl_version:
+		// the logic comes from exhaustive search and matching between OpenGL and Vulkan GPUInfo website
+		// https://vulkan.gpuinfo.org/displaycoreproperty.php?core=1.2&name=driverID&platform=all
+		if(hasInString(vendor, "mesa", true) || hasInString(renderer, "mesa", true) || hasInString(ogl_ver_str, "mesa", true))
+		{
+			// Mesa Driver
+			if(hasInString(vendor, "intel", true) || hasInString(renderer, "intel", true))
+				m_properties.driverID = E_DRIVER_ID::EDI_INTEL_OPEN_SOURCE_MESA;
+			else if(hasInString(renderer, "radv", true))
+				m_properties.driverID = E_DRIVER_ID::EDI_MESA_RADV;
+			else if(hasInString(renderer, "llvmpipe", true))
+				m_properties.driverID = E_DRIVER_ID::EDI_MESA_LLVMPIPE;
+			else if(hasInString(renderer, "amd", true) || hasInString(renderer, "radeon", true) || hasInString(vendor, "ati technologies", true))
+				m_properties.driverID = E_DRIVER_ID::EDI_AMD_OPEN_SOURCE;
+		}
+		else if(hasInString(vendor, "intel", true) || hasInString(renderer, "intel", true))
+			m_properties.driverID = E_DRIVER_ID::EDI_INTEL_PROPRIETARY_WINDOWS;
+		else if(hasInString(vendor, "ati technologies", true) || hasInString(vendor, "amd", true) || hasInString(renderer, "amd", true))
+			m_properties.driverID = E_DRIVER_ID::EDI_AMD_PROPRIETARY;
+		else if(hasInString(vendor, "nvidia", true)) // easiest to detect :D
+			m_properties.driverID = E_DRIVER_ID::EDI_NVIDIA_PROPRIETARY;
+		else 
+			m_properties.driverID = E_DRIVER_ID::EDI_OTHER;
+
+		m_glfeatures.isIntelGPU = (m_properties.driverID == E_DRIVER_ID::EDI_INTEL_OPEN_SOURCE_MESA || m_properties.driverID == E_DRIVER_ID::EDI_INTEL_PROPRIETARY_WINDOWS);
+
+		// Heuristic to detect Physical Device Type until we have something better:
+		if(m_properties.driverID == E_DRIVER_ID::EDI_INTEL_OPEN_SOURCE_MESA || m_properties.driverID == E_DRIVER_ID::EDI_INTEL_PROPRIETARY_WINDOWS)
+			m_properties.deviceType = E_TYPE::ET_INTEGRATED_GPU;
+		else if(m_properties.driverID == E_DRIVER_ID::EDI_AMD_OPEN_SOURCE || m_properties.driverID == E_DRIVER_ID::EDI_AMD_PROPRIETARY)
+			m_properties.deviceType = E_TYPE::ET_DISCRETE_GPU;
+		else if(m_properties.driverID == E_DRIVER_ID::EDI_NVIDIA_PROPRIETARY)
+			m_properties.deviceType = E_TYPE::ET_DISCRETE_GPU;
+		else if(hasInString(renderer, "virgl", true))
+			m_properties.deviceType = E_TYPE::ET_VIRTUAL_GPU;
+		else
+			m_properties.deviceType = E_TYPE::ET_OTHER;
```

Comment:
you can also use `#if defined(NBL_PLATFORM)` to help yourself differentiate between WINDOWS and others